File: app/src/main/java/com/wya/env/net/BaseSubscriber.java
package com.wya.env.net;

import io.reactivex.Observer;
import io.reactivex.disposables.Disposable;
import com.wya.env.base.BaseActivity;
import com.wya.env.base.BaseView;

/**
 * Created by Administrator on 2018/5/31 0031.
 * 写一个Subscriber继承Observer，重写相应的方法，在具体调用时，只需要重写你需要的onNext方法即可，不需要写入其余的回调方法
 */

public class BaseSubscriber<T> implements Observer<T> {
    private BaseView mBaseView;
    private Disposable disposable;
    public BaseSubscriber(BaseView baseView) {
        this.mBaseView=baseView;
    }

    @Override
    public void onSubscribe(Disposable d) {
        disposable = d;
    }

    @Override
    public void onNext(T t) {
        mBaseView.hideLoading();
    }

    @Override
    public void onError(Throwable e) {
        mBaseView.hideLoading();
        mBaseView.failedResult(e.getMessage());
    }

    @Override
    public void onComplete() {
        mBaseView.hideLoading();
    }
}

File: app/src/main/java/com/wya/env/http/BaseSubscriber.kt
package com.wya.env.http

import com.wya.env.base.BaseView
import io.reactivex.Observer
import io.reactivex.disposables.Disposable

/**
 * @date: 2018/5/31 13:59
 * @author: Chunjiang Mao
 * @classname: BaseSubscriber
 * @describe: 写一个Subscriber继承Observer，重写相应的方法，在具体调用时，只需要重写你需要的onNext方法即可，不需要写入其余的回调方法
 */

open class BaseSubscriber<T>(private val mBaseView: BaseView) : Observer<T> {
    private var disposable: Disposable? = null

    override fun onSubscribe(d: Disposable) {
        disposable = d
    }

    override fun onNext(t: T) {
        mBaseView.hideLoading()
    }

    override fun onError(e: Throwable) {
        mBaseView.hideLoading()
        mBaseView.failedResult(e.message!!)
    }

    override fun onComplete() {
        mBaseView.hideLoading()
    }
}
